### PR TITLE
Evaluate constants in array repeat expression

### PR DESCRIPTION
**Pull request by HKalbasi**
cc #8655

### CODE REVIEW

File: crates/ide_assists/src/handlers/add_explicit_type.rs [hunk 1]
```diff
@@ -180,14 +180,17 @@ fn main() {
 }
 "#,
         );
-        // note: this may break later if we add more consteval. it just needs to be something that our
-        // consteval engine doesn't understand
-        check_assist_not_applicable(
+        check_assist(
```
Comment (flodiebold) [inline crates/ide_assists/src/handlers/add_explicit_type.rs:183]: this is supposed to test a case where the assist isn't applicable, so would you mind changing it to something else and leave the comment?

**Review by flodiebold — APPROVED** [review]
bors d+

File: crates/ide_assists/src/handlers/add_explicit_type.rs (hunk 1 above)
Comment (HKalbasi) [reply crates/ide_assists/src/handlers/add_explicit_type.rs]: Ah I missed the comment's point completely

**Comment by lnicola** [comment]
image

**Pull request merged**